### PR TITLE
Add tag support to pyramid_metrics.

### PR DESCRIPTION
**Pull request by jcsackett**

### CODE REVIEW

**Review by makyo — APPROVED** [review]

File: pyramid_metrics/utility.py [hunk 1]
```diff
@@ -86,27 +86,29 @@ def _key(self, stat):
         else:
             return str(stat)
 
-    def incr(self, stat, count=1, per_route=False):
+    def incr(self, stat, count=1, tags={}, per_route=False):
```
Comment (frankban) [inline pyramid_metrics/utility.py]: Mutable kwarg.

File: pyramid_metrics/utility.py [hunk 2]
```diff
@@ -86,27 +86,29 @@ def _key(self, stat):
...
-        self._statsd.incr(self._key(stat), count=count)
+        self._statsd.incr(self._key(stat), count=count, tags=tags)
         if per_route:
-            self._statsd.incr(self._route_key(self._key(stat)),
-                              count=count)
+            self._statsd.incr(
+                self._route_key(self._key(stat)), count=count, tags=tags)
 
-    def gauge(self, stat, value, delta=False, per_route=False):
+    def gauge(self, stat, value, delta=False, tags={}, per_route=False):
```
Comment (frankban) [inline pyramid_metrics/utility.py]: Mutable kwarg.

File: pyramid_metrics/utility.py [hunk 3]
```diff
@@ -86,27 +86,29 @@ def _key(self, stat):
...
+        self._statsd.gauge(self._key(stat), value, tags=tags, delta=delta)
         if per_route:
-            self._statsd.gauge(self._route_key(self._key(stat)), value,
-                               delta=delta)
+            self._statsd.gauge(
+                self._route_key(
+                    self._key(stat)), value, tags=tags, delta=delta)
 
-    def timing(self, stat, dt, per_route=False):
+    def timing(self, stat, dt, tags={}, per_route=False):
```
Comment (frankban) [inline pyramid_metrics/utility.py]: Mutable kwarg.

File: pyramid_metrics/utility.py [hunk 4]
```diff
@@ -86,27 +86,29 @@ def _key(self, stat):
...
         """ Push a TIMER metric """
-        self._statsd.timing(self._key(stat), int(dt * 1000))
+        self._statsd.timing(self._key(stat), int(dt * 1000), tags=tags)
         if per_route:
-            self._statsd.timing(self._route_key(self._key(stat)), int(dt * 1000))
+            self._statsd.timing(
+                self._route_key(self._key(stat)), int(dt * 1000), tags=tags)
 
-    def timer(self, stat, per_route=False):
+    def timer(self, stat, tags={}, per_route=False):
```
Comment (frankban) [inline pyramid_metrics/utility.py]: Ditto.

File: pyramid_metrics/utility.py [hunk 5]
```diff
@@ -124,7 +126,7 @@ def mark_start(self, stat):
         start_marker = Marker(marker_name)
         self.active_markers[marker_name] = start_marker
 
-    def mark_stop(self, stat, prefix='', suffix='', per_route=True):
+    def mark_stop(self, stat, prefix='', suffix='', tags={}, per_route=True):
```
Comment (frankban) [inline pyramid_metrics/utility.py]: Ditto.

**Review by frankban — APPROVED** [review]
Looks good with changes.

**Pull request merged**